### PR TITLE
BREAKING: container > navigation

### PR DESCRIPTION
**Pull request by rjrjr**
In preparation for (finally!) locking down the workflow-ui API, renames the `com.squareup.workflow1.ui.container` package to `navigation`, since that's the common theme of nearly everything in there. I think this simple change should help make our navigation story a bit more self documentating. Yes, I'll still write navigation docs.

Other changes:

 * `View.setBackHandler` moves from `ui` to `ui.navigation`
 * `EnvironmentScreen`, `Screen.withEnvironment()`, `Screen.withRegistry()` which this PR moves up to the `ui` package. References to these had bled out into other spots already, so this feels like a move in the right direction.
 * `OnBackPressedDispatcherOwnerKey` moves from `ui` to `ui.androidx`

Also, note that the packages of the deprecated classes in the `container-android` and `container-common` modules were not changed, since they're about to be deleted.

### CODE REVIEW

File: samples/containers/poetry/src/main/java/com/squareup/sample/poetry/StanzaListScreen.kt [hunk 1]
```diff
@@ -16,9 +16,9 @@ import com.squareup.workflow1.ui.ScreenViewFactory
 import com.squareup.workflow1.ui.ScreenViewRunner
 import com.squareup.workflow1.ui.ViewEnvironment
 import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
-import com.squareup.workflow1.ui.container.BackStackConfig
-import com.squareup.workflow1.ui.container.BackStackConfig.Other
-import com.squareup.workflow1.ui.setBackHandler
+import com.squareup.workflow1.ui.navigation.BackStackConfig
+import com.squareup.workflow1.ui.navigation.BackStackConfig.Other
+import com.squareup.workflow1.ui.container.setBackHandler
```
Comment (rjrjr) [inline samples/containers/poetry/src/main/java/com/squareup/sample/poetry/StanzaListScreen.kt:21]: Missed some spots :/

**Review by ekeitho — APPROVED** [review]
LGTM

**Comment by rjrjr** [comment]
Force push fixes file name of `ViewSetBackHandler.kt` and improves commit message.

File: samples/nested-overlays/src/main/java/com/squareup/sample/nestedoverlays/NestedOverlaysWorkflow.kt [hunk 2]
```diff
@@ -53,11 +55,11 @@ object NestedOverlaysWorkflow : StatefulWorkflow<Unit, State, Nothing, Screen>()
         BackButtonScreen(
           ButtonBar(
             Button(
-              name = R.string.close,
+              name = string.close,
```
Comment (ijwhelan) [inline samples/nested-overlays/src/main/java/com/squareup/sample/nestedoverlays/NestedOverlaysWorkflow.kt:58]: Android Studio strikes again

Comment (ijwhelan) [reply samples/nested-overlays/src/main/java/com/squareup/sample/nestedoverlays/NestedOverlaysWorkflow.kt:58]: Looks like it's only this file

**Review by ijwhelan — APPROVED** [review]
Overall LGTM, just one place where Android Studio tried to be too helpful

File: samples/nested-overlays/src/main/java/com/squareup/sample/nestedoverlays/NestedOverlaysWorkflow.kt (hunk 2 above)
Comment (rjrjr) [reply samples/nested-overlays/src/main/java/com/squareup/sample/nestedoverlays/NestedOverlaysWorkflow.kt:58]: Thanks for catching that. It's just crazy making.

**Pull request merged**